fitbenchmarking/core/results_output.py

```diff
@@ -1,151 +1,233 @@
+    for each problem
+
+    :param results_per_test: results nested array of objects
+    :type results_per_test : list of list of
+                             fitbenchmarking.utils.fitbm_result.FittingResult
+
+    :return: The best result for each problem
+    :rtype: list of fitbenchmarking.utils.fitbm_result.FittingResult
+    """
+    output = []
+    for results in results_per_test:
+        best_result = min(results, key=lambda x: x.chi_sq)
+        min_chi_sq = best_result.chi_sq
+        min_runtime = min([r.runtime for r in results])
+        for r in results:
+            r.min_chi_sq = min_chi_sq
+            r.min_runtime = min_runtime
+            r.set_colour_scale()
+        output.append(best_result)
+    return output
 
-    software = options.software
-    if not isinstance(software, list):
-        software = [software]
-    minimizers = [options.minimizers[s] for s in software]
-    minimizers = sum(minimizers, [])
 
-    results_dir = options.results_dir
-    use_errors = options.use_errors
+def create_plots(options, results, best_results, group_name, figures_dir):
+    """
+    Create a plot for each result and store in the figures directory
+
+    :param options: The options used in the fitting problem and plotting
+    :type options: fitbenchmarking.utils.options.Options
+    :param results: results nested array of objects
+    :type results : list of list of
+                    fitbenchmarking.utils.fitbm_result.FittingResult
+    :param best_results: best result for each problem
+    :type best_results: list of
+                        fitbenchmarking.utils.fitbm_result.FittingResult
+    :param group_name: name of he problem group
+    :type group_name: str
+    :param figures_dir: Path to directory to store the figures in
+    :type figures_dir: str
+    """
+    name_count = {}
+    for best, prob_result in zip(best_results, results):
+        name = best.problem.name
+        name_count[name] = 1 + name_count.get(name, 0)
+        count = name_count[name]
+
+        plot = plots.Plot(problem=best.problem,
+                          options=options,
+                          count=count,
+                          figures_dir=figures_dir)
+
+        # Create a plot showing the initial guess and get filename
+        initial_guess_path = plot.plot_initial_guess()
+
+        # If none of the fits succeeded, params could be None
+        # Otherwise, add the best fit to the plot
+        if best.params is not None:
+            plot.plot_best(best.minimizer, best.params)
+
+        # For each result, if it succeeded, create a plot and add plot links to
+        # the resuts object
+        for result in prob_result:
+            if result.params is not None:
+                plot_path = plot.plot_fit(result.minimizer, result.params)
+                result.figure_link = plot_path
+            else:
+                result.figure_link = 'Failed to return params'
```

Comment:
Similarly to the make_plots off, is it possible to remove the directory path and produce the full message?
![Screenshot at 2020-01-15 14-08-09](https://user-images.githubusercontent.com/15614849/72440305-7d9da580-37a0-11ea-8a5e-b65371fe8145.png)

Reply:
Done